crypto/err/err.c

```diff
@@ -146,13 +147,51 @@ static int err_string_data_cmp(const ERR_STRING_DATA *a,
     return a->error > b->error ? 1 : -1;
 }
 
+DEFINE_RUN_ONCE_STATIC(do_err_lock_init)
+{
+    return (err_lock = CRYPTO_THREAD_lock_new()) != NULL;
+}
+
+DEFINE_RUN_ONCE_STATIC(do_err_strings_init)
+{
+    int_error_hash = lh_ERR_STRING_DATA_new(err_string_data_hash,
+                                            err_string_data_cmp);
+    return int_error_hash != NULL;
+}
+
+DEFINE_RUN_ONCE_STATIC(do_err_local_init)
+{
+    set_err_thread_local = 1;
+    return CRYPTO_THREAD_init_local(&err_thread_local, NULL);
+}
+
+static int err_local_init(void)
+{
+    return OPENSSL_init_crypto(0, NULL)
```

Comment:
Why is this needed? My assumption is we need to init threads because of the thread local storage. There needs to be a comment against all of these explaining it. I think it is probably better to auto-initialise threading from inside the `CRYPTO_THREAD_init_local` call.